src/openrct2/game.c

```diff
@@ -1357,8 +1359,8 @@ bool game_load_save_or_scenario(const utf8 * path)
 
 static void game_load_or_quit_no_save_prompt_callback(sint32 result, const utf8 * path)
 {
-    if (result == MODAL_RESULT_OK && game_load_save_or_scenario(path)) {
-        gFirstTimeSave = 0;
+    if (result == MODAL_RESULT_OK) {
+        game_load_save_or_scenario(path);
```

Comment:
with the edits i did in the game_load_save(path), game_load_save_or_scenario now handles setting the gFirstTimeSaving on a per-file basis. https://github.com/wolfreak99/OpenRCT2/blob/0dcbacb09b52f96382159d325a63abcb265f9644/src/openrct2/game.c#L1084-L1091

The original method caused sc6 files to mark gFirstTimeSave to be true during game_load_save_or_scenario, but then reset them after exiting game_load_save_or_scenario.